parm/config/config.resources

```diff
@@ -67,72 +73,88 @@ elif [ ${step} = "waveinit" ]; then
     export wtime_waveinit="00:10:00"
     export npe_waveinit=12
     export nth_waveinit=1
-    export npe_node_waveinit=$(echo "${npe_node_max} / ${nth_waveinit}" | bc)
+    export npe_node_waveinit=$npe_waveinit
```

Comment:
Won't some of these change if the number of threads is increased later?

Reply:
Some of these were brought in from the GFSv16 WCOSS2 port. I reviewed the `npe_node_*` variables and made adjustments. These adjustments did not change the xml resources for a high-res cycled test on WCOSS2 so that's good. The wave jobs aren't run in testing for this PR so we'll need to check them when we turn waves back on in the future. Similar note for the wafs jobs.

I also checked that the xml resources don't change on Hera with these updates. They don't.

I also pulled in recent updates from GFSv16.3 (ecf scripts and memory reductions).

Please review the `config.resources` changes.